cases/src/org/commcare/cases/util/CasePurgeFilter.java

```diff
@@ -119,21 +123,22 @@ private void setIdsToRemoveWithNewExtensions(IStorageUtilityIndexed<Case> caseSt
                 nodeStatus |= STATUS_RELEVANT;
             }
 
-            graph.addNode(c.getCaseId(), new int[]{nodeStatus, c.getID()});
+            internalCaseDAG.addNode(c.getCaseId(), new int[]{nodeStatus, c.getID()});
 
             for (CaseIndex index : indexHolder) {
-                graph.setEdge(c.getCaseId(), index.getTarget(), index.getRelationship());
+                internalCaseDAG.setEdge(c.getCaseId(), index.getTarget(), index.getRelationship());
             }
             indexHolder.removeAllElements();
         }
 
-        propogateRelevance(graph);
-        propogateAvailabile(graph);
-        propogateLive(graph);
+        invalidEdgesWereRemoved = removeInvalidEdges();
```

Comment:
@ctsims Not sure I understand exactly what you're looking for here, but I think I addressed this by renaming removeInvalidEdges() to getInvalidEdges(), and having it return the list of edges to remove, and then having the caller method loop through and do the actual removal of edges from the DAG.